### PR TITLE
Solve #391, Refactor Rule into a submodule

### PR DESCRIPTION
**Pull request by littlezhou**

### CODE REVIEW

File: smart-rule/src/test/resources/accessCountTable.xml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        <row>
+            <value>104</value>
+            <value>5</value>
+        </row>
+        <row>
+            <value>105</value>
+            <value>4</value>
+        </row>
+    </table>
+</dataset>
```
Comment (huafengw) [inline smart-rule/src/test/resources/accessCountTable.xml]: Is this one related?

Comment (littlezhou) [inline smart-rule/src/test/resources/accessCountTable.xml]: O, not related at all, copied to the module by mistake.

**Comment by drankye** [comment]
Would be good to rename some classes by the way of this change, like File -> FileObject.
